### PR TITLE
add swap sim example

### PR DESCRIPTION
**Pull request by Evan-Kim2028**
extracted `test_swap_sim` test into an example file

### CODE REVIEW

**Comment by Evan-Kim2028** [comment]
Not sure why the test suite is failing, I didn't touch any code that should affect those tests. 

Regarding  rustfmt fail, I adjusted the doc lines to conform to rustfmt line size

**Comment by Evan-Kim2028** [comment]
Ran  `cargo fmt` and `cargo clippy --fix`

File: examples/calculate-price.rs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    IQuoter,
+r#"[
+    function quoteExactInputSingle(address tokenIn, address tokenOut,uint24 fee, uint256 amountIn, uint160 sqrtPriceLimitX96) external returns (uint256 amountOut)
+]"#;);
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error>> {
+    // load rpc endpoint from local environment
+    let rpc_endpoint = std::env::var("ETHEREUM_MAINNET_ENDPOINT")
+        .expect("Could not get ETHEREUM_MAINNET_ENDPOINT");
```
Comment (0xKitsune) [inline examples/calculate-price.rs:22]: Since you are using Box<dyn Error>, you can just use `?` instead of `expect()`.

File: examples/calculate-price.rs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    let provider = Arc::new(Provider::<Http>::try_from(rpc_endpoint).unwrap());
+
+    //Instantiate Pools and Quoter
+    let mut pool = UniswapV3Pool::new_from_address(
+        H160::from_str("0x88e6A0c2dDD26FEEb64F039a2c41296FcB3f5640").unwrap(), // univ3 usdc/weth pool
+        provider.clone(),
+    )
+    .await // use ? at end of await??
+    .unwrap();
```
Comment (0xKitsune) [inline examples/calculate-price.rs:32]: Lets use `?` instead of `unwrap()`

File: examples/calculate-price.rs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    pool.get_pool_data(provider.clone()).await.unwrap();
+
+    let block_pool = IUniswapV3Pool::new(
+        H160::from_str("0x88e6A0c2dDD26FEEb64F039a2c41296FcB3f5640").unwrap(),
+        provider.clone(),
+    );
+
+    let current_block = provider.get_block_number().await.unwrap();
```
Comment (0xKitsune) [inline examples/calculate-price.rs:41]: Same thing here, lets use `?`

File: examples/calculate-price.rs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+    let sqrt_price = block_pool
+        .slot_0()
+        .block(current_block)
+        .call()
+        .await
+        .unwrap()
```
Comment (0xKitsune) [inline examples/calculate-price.rs:48]: Same as above, use `?` instead.

File: examples/calculate-price.rs [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        .0;
+    pool.sqrt_price = sqrt_price;
+
+    let float_price_a = pool.calculate_price(pool.token_a);
+
+    let float_price_b = pool.calculate_price(pool.token_b);
+
+    dbg!(pool);
```
Comment (0xKitsune) [inline examples/calculate-price.rs:56]: Can you remove the `dbg!()` from this?

File: examples/sim-swap.rs [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    IQuoter,
+r#"[
+    function quoteExactInputSingle(address tokenIn, address tokenOut,uint24 fee, uint256 amountIn, uint160 sqrtPriceLimitX96) external returns (uint256 amountOut)
+]"#;);
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error>> {
+    // load rpc endpoint from local environment
+    let rpc_endpoint = std::env::var("ETHEREUM_MAINNET_ENDPOINT")
+        .expect("Could not get ETHEREUM_MAINNET_ENDPOINT");
```
Comment (0xKitsune) [inline examples/sim-swap.rs:21]: Same as in the `calculate-price` example, lets use `?` everywhere instead of `unwrap()` or `expect()`.

File: src/pool/mod.rs [hunk 7]
```diff
@@ -40,14 +47,16 @@ impl Pool {
         }
     }
 
+    /// Returns the fee percentage associated with the pool.
```
Comment (0xKitsune) [inline src/pool/mod.rs:50]: Can we reword this to "Returns the fee associated with the pool, represented as a percentage?

**Comment by 0xKitsune** [comment]
Thanks for making this PR! I think we should probably separate the docs into a separate PR. I think that there are a few places where some additional information and rewording could be useful, but we can get the examples merged in once the changes requested are made.
